src/main/java/com/amazonaws/services/kinesis/connectors/KinesisConnectorRecordProcessor.java

```diff
@@ -126,8 +126,18 @@ public void processRecords(List<Record> records, IRecordProcessorCheckpointer ch
                 } else {
                     throw new RuntimeException("Transformer must implement ITransformer or ICollectionTransformer");
                 }
-            } catch (IOException e) {
+            } catch (Exception e) {
                 LOG.error(e);
+                try {
+                    final String lastSequenceNumberProcessed = buffer.getLastSequenceNumber();
```

Comment:
Currently, checkpoint is done when the record(s) have been processed successfully (in the emit method).  But in case of Exceptions in any of the stages, (transform->filter->buffer->emit ), checkpoint is not done. For example, if an exception is thrown due to bad data in a particular record, checkpoint is never done and hence, the same record will be processed over and over again raising the same Exception. Therefore, the consumer will never move forward in the processing of records. It appeared to happen in my application.